### ISSUE TITLE
Updates on official documentation

### ISSUE THREAD

**Issue by Zerotistic** [Vector35/binaryninja-api#4689]
**Is this a general concept that needs to be documented or a specific API?**
It is general concepts and updates to already documented section.

**What concept or API needs more documentation?**
I am sorry for this is many different parts/sections
- [x] How to install the API (maybe link to https://docs.binary.ninja/dev/batch.html#install-the-api ? or mention its necessary to install it separetely): https://docs.binary.ninja/getting-started.html#linux
- [x] Its an old pic, not much has changed but imo it looks better now: https://docs.binary.ninja/getting-started.html#ui-basics
- [x] Same here, old pic, but it has changed with many new options: https://docs.binary.ninja/guide/index.html#loading-files
- [x] Explain what are Auto Tags: https://docs.binary.ninja/dev/annotation.html#auto-tags
- [x] Clean the output of `TypePrinter.print_all_types` (unless that's intended, but I would suggest it nonetheless): https://docs.binary.ninja/dev/annotation.html#headers
- [x] Add description of new plugin examples: https://github.com/Vector35/binaryninja-api/tree/dev/python/examples
- [x] Write the basic concepts of workflows: https://docs.binary.ninja/dev/workflows.html#basic-concepts
- [x] I believe a section similar to "more resources" would be nice in the getting started. You could (at least) include links to your trainings (https://binary.ninja/training/), videos (https://www.youtube.com/watch?v=xKBQatwshs0&list=PLCVV6Y9LmwOgqqT5obf0OmN9fp5495bLr&index=1) and mention Jordan's (irregular but sick) live: https://docs.binary.ninja/getting-started.html
- [x] UI plugin should either have a template or a section dedicated to them as its a more complex usage (i personally struggled a lot with it): https://docs.binary.ninja/dev/plugins.html#ui-plugins
- [x] Talk about SSA in the "Important Concepts" category: https://docs.binary.ninja/dev/concepts.html
- [x] The fix seems to have been made, don't think its worth mentioning it anymore? https://docs.binary.ninja/dev/documentation.html#prerequisites
- [x] Adding how to rebase the binary with the API would be nice, its not hard but very useful: https://docs.binary.ninja/dev/batch.html

**Is this documentation related to a specific API?**
No it is not.

**Are there any known examples of people using this API/concept?**
Aside from myself, there are people who made community plugins and I had colleagues ask me.

**Comment by psifertex** [comment]
It's not actually necessary to install separately -- the linux setup script calls the other one for you automatically.
https://github.com/Vector35/binaryninja-api/blob/dev/scripts/linux-setup.sh#L72

I'm going to make a [separate issue](https://github.com/Vector35/binaryninja-api/issues/4690) for all of the images as _every_ image in the documentation and website will be redone for 4.0 as we're changing the default theme.

`print_all_types` is intended, but it's easy enough to improve, done. 

More descriptions written, thanks!

Workflows unfortunately needs to hold off until we've made some important internal changes that will impact how they're used. Once the updated feature lands (hopefully in 4.0 it should include documentation)

Since Rebasing is a single API call that comes up when searching for that keyword I don't think calling it out specifically is really helpful. Though it could use an example in the API docs which I've just added! 



The other resources I added to the "what's next" section of the getting started guide.

**Comment by psifertex** [comment]
The only things not yet resolved are the UI plugin details or the SSA documentation as those will take a bit more time. I've actually already been working on updated documentation for UI plugins, but it would really help me would be if you could describe:

- What did you want to do with a UI plugin that didn't come up with an API search? 
- Which examples were the most helpful?
- Are there some APIs you think need more examples than others? 

RE: SSA form, we should probably just write up a more formal description of SSA in https://docs.binary.ninja/dev/bnil-overview.html or as a separate page in that section and then link to important concepts while demonstrating how to walk the SSA forms in a particular example

**Comment by Zerotistic** [comment]
I was planning to do a PR to fix most of those but you beat me to it!

> It's not actually necessary to install separately -- the linux setup script calls the other one for you automatically. https://github.com/Vector35/binaryninja-api/blob/dev/scripts/linux-setup.sh#L72

Oh is it not? I remember having to do it manually before, maybe its because I am on an unsupported distro (obligatory I use arch btw)?

> I'm going to make a [separate issue](https://github.com/Vector35/binaryninja-api/issues/4690) for all of the images as _every_ image in the documentation and website will be redone for 4.0 as we're changing the default theme.

Yeah I just saw your issue and was going to comment you should track those there.

> Workflows unfortunately needs to hold off until we've made some important internal changes that will impact how they're used. Once the updated feature lands (hopefully in 4.0 it should include documentation)

Fair enough, no need to update the doc if it'll be updated again soon. Excited for v4.0 👀 

> The only things not yet resolved are the UI plugin details or the SSA documentation as those will take a bit more time. 
Fair enough, completely understand.
> I've actually already been working on updated documentation for UI plugins, but it would really help me would be if you could describe:
> 
>     * What did you want to do with a UI plugin that didn't come up with an API search?
> 
>     * Which examples were the most helpful?
> 
>     * Are there some APIs you think need more examples than others?

It is likely that I just don't know what word to look up, but "Sidebar" in the API search only brings back color related functions. I tried "Widget', which only bring back color related functions too. Looking at the navbar for "UI" or something similar there is no result (on the Python API doc). 
I believe I had to go look at the community plugins to figure it out. If not adding documentation for it, at least somehow mention what and where we should look for in the API?
Searching how to update the new widget (I think its like that you call them?) or how to have different data in it depending on the view had me struggle a lot. 

The most useful resources were community plugins and Qt6 documentation (and a bit of trial and error).

I can't think of any right now, but I can come back to you later if you want. I haven't had to make an UI in a while.

**Comment by psifertex** [comment]
Ahh, yeah, the main issue there is that all of the UI stuff is only documented in the [C++ docs](https://api.binary.ninja/cpp/) due to some Shiboken quirks. The more I think about it the more a small update to the UI plugin documentation could help. Let me take a stab at it. I'd still welcome any PRs if you see other things though!

**Comment by psifertex** [comment]
Final items are done with https://github.com/Vector35/binaryninja-api/commit/66be2657a30056b99fab83c7211092f288b44e69
